### ISSUE TITLE
Unable to connect websocket on Iframe

### ISSUE THREAD

**Issue by rajatheynetech** [Ylianst/MeshCentral#4463]
Remote desktop embedding on iframe shows "Unable to connect to WebSocket -- reconnect". But the same URL on iframe src is working perfectly on the browser.

Iframe screen
<img width="1290" alt="Screenshot 2022-08-29 at 2 25 05 PM" src="https://user-images.githubusercontent.com/67583922/187174383-f8b2ca38-2405-4e90-bc6c-fa94cf6e7b45.png">

**Comment by Ylianst** [comment]
Can you open your browser's console and show any errors?

Also, did you put:

```
"allowframing": true
```

In the settings section of your config.json?

**Comment by si458** [comment]
@Ylianst i believe the user did #4461

**Comment by rajatheynetech** [comment]
> Can you open your browser's console and show any errors?
> 
> Also, did you put:
> 
> ```
> "allowframing": true
> ```
> 
> In the settings section of your config.json?

> Yes i did

**Comment by rajatheynetech** [comment]
<img width="1440" alt="Screenshot 2022-08-30 at 2 44 07 PM" src="https://user-images.githubusercontent.com/67583922/187398696-c333df97-05e5-4561-a4d6-749c4fb71f67.png">

**Comment by rajathdeskday** [comment]
> @Ylianst i believe the user did #4461

This issue was raised by me. And the current issue is not with the iframe. The URL loaded in the iframe shows an error like WebSocket couldn't connect.

**Comment by sufailDesk** [comment]
Is there a fix for this problem? A 404 error is returned by the commander.ashx while trying to embed in an iframe. However, there is no problem when using the same url directly in the browser.

**Comment by silversword411** [comment]
@rajatheynetech ...you're trying to connect to `127.0.0.1` instead of the dns name, so you're going to have SSL negotiation problems.

SSL requires the right names in the certs in the browser.

**Comment by sufailDesk** [comment]
No, it is connecting to htpps://domain.com/commander.ashx

**Comment by sufailDesk** [comment]
Like I mentioned above the problem is only for remote connecting through embedding into an iframe. Embedding is done by following only meshcentral's user guide.  No issue when the same url directly applied into the browser.

**Comment by silversword411** [comment]
I understand you're trying to use an iframe and embed it in your app. This is your user guide's instructions right?

https://ylianst.github.io/MeshCentral/meshcentral/#embedding-meshcentral

I see no references to any "commander.ashx" What is that? I only see commander.htm in MeshCentral's source code: https://github.com/Ylianst/MeshCentral/blob/master/public/commander.htm

Your error is also about FLoC (which is dead and being removed from chrome), so you must be in a special dev mode to even see that error. https://www.drupal.org/project/drupal/issues/3284706

**Comment by dkapadiaatlivewire** [comment]
@Ylianst @rajatheynetech  did you get resolution of this error while running MeshCentral in Iframe?

**Comment by sufailDesk** [comment]
> @Ylianst @rajatheynetech did you get resolution of this error while running MeshCentral in Iframe?

No. Still not work for iframe. but we have not tried the latest solutions which mentioned above.

**Comment by bindulu05** [comment]
@silversword411  I am facing the same issue in iframe when I am trying to connect to MeshCentral URL with login token .Without login token it is working fine it is displaying login page of meshcentral

**Comment by King-tu** [comment]
> Remote desktop embedding on iframe shows "Unable to connect to WebSocket -- reconnect". But the same URL on iframe src is working perfectly on the browser.
> 
> Iframe screen <img alt="Screenshot 2022-08-29 at 2 25 05 PM" width="1290" src="https://user-images.githubusercontent.com/67583922/187174383-f8b2ca38-2405-4e90-bc6c-fa94cf6e7b45.png">

I encountered the same problem and finally found that the error was caused by cross-site request. I solved this problem by setting the "sessionSameSite": "None" configuration item in config.json.
Refer to the following issue:
[Meshcentral not loading in iFrame in Chrome or Edge | Missing SameSite Attribute on Cookie #4211](https://github.com/Ylianst/MeshCentral/issues/4211).
